backend/src/lib/database/db.rs

```diff
@@ -118,17 +127,30 @@ impl DatabaseClient {
         &self,
         login_details: &LoginDetails,
     ) -> Result<String, LoginError> {
-        // Find the password in the database by email or username
-        let hashed_password = self.get_password_for_email(&login_details.email).await;
+        // Determine if username or password was used
+        // Right now the Ok below is unused, only really checking the error
+        let login_method = match (&login_details.email, &login_details.username) {
+            (Some(_email), _) => Ok(LoginMethod::Email),
+            (None, Some(_username)) => Ok(LoginMethod::Username),
+            (None, None) => Err(LoginError::MissingCredentials),
+        }?;
+
+        // Get user data
+        let user_details = self
+            .get_details_by_login_method(&login_method, login_details)
+            .await;
 
         // If no password is found for the username or email, the user does not exist
-        if hashed_password.is_err() {
-            log::error!("User does not exist");
+        if user_details.is_err() {
+            log::error!("User does not exist!");
             return Err(LoginError::UserDoesntExist);
         };
```

Comment:
I think you should just return the error that you got from the `.get_details_by_login_method()` instead of creating a new one here:
```rust
        // Get user data
        let user_details = self
            .get_details_by_login_method(&login_method, login_details)
            .await?;
```